sessions/TypeScript-100/Module-102-common-types/common-types.md

```diff
@@ -0,0 +1,81 @@
+
+# Module 102: Common Types
+
+## Primitives
+
+Along with this file, there are code examples of each type in [common-types-examples.ts](./common-types-examples.ts).
+
+
+Similar to JavaScript, there are 3 commonly used primitives
+- `boolean`
+	- Same as JavaScript, simple true/false value
+- `number`
+    - There is no int or float, everything is number to JavaScript /TS
+- `string`
+	- Like JavaScript , strings can use single quotes or double quotes    
+	- Template string using backticks and the ${} syntax
+- `symbol `
+	- There is a primitive in JavaScript used to create a globally unique reference via the function Symbol()
+	- Symbol represenTypeScript unique tokens that can be used as keys for object properties.
+- `bigint` 
+	- Introduced in TypeScript 3.2, it provides a way to represent whole numbers larger than 253.
+	- You can get a bigint in 2 ways:
+		- Calling the BigInt() function    
+		- Writing a BigInt literal by adding `n` to the end of any numeric integer
+	- NOTE: You can use the bigint only if you are targeting version ESNext.
+- `null`
+	- Like JavaScript, null is used to indicate an absent value
+- `undefined`
+	- Like JavaScript, undefined is used to indicate an uninitialised value
+
+Note: `null` and `undefined` depend on whether you have the [strictNullChecks](https://www.typescriptlang.org/tsconfig#strictNullChecks) option on (this will be covered in a later module)
+- `any`
+	- A top type of TypeScript's type system (aka universal supertype)  
+	- This type allows TypeScript to work with existing JavaScript , with any you can essentially opt-out of type checking during compilation.
+	- any represents all possible JavaScript values; primitives, objects, arrays, functions, errors, symbols, etc
+	- As developers, this gives us a ton of freedom: TypeScript lets us perform any operation we want on values of type any without having to perform any kind of checking beforehand.
+	- In most cases, this is too permissive and can be problematic at runtime. There is not much protection offered from TypeScript when using the any type
+- `unknown`
+	- Another top type of TypeScript's type system
+	- If a variable is not known at the time of writing (e.g. accepting all values from user input) we can use the unknown type here
+	- You can narrow your unknown type variable to something more specific by doing typeof checks, comparison checks, or more advanced type guards
+	- This is similar to any, as the value can be of any type but it is essentially safe by default
+- `objects`
```

Comment:
Wonder if it's worth calling out the difference between `object` and `Object` ?

https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#number-string-boolean-symbol-and-object

Reply:
That's a good shout, there is a document which is essentially the speaker notes. I'll add something in this file for the do's & don'ts of TS with the link to the handbook

Reply:
decided to add it very early in the file to bring it to attention sooner!